sklearn/tests/test_pipeline.py

```diff
@@ -330,6 +332,40 @@ def test_pipeline_methods_pca_svm():
     pipe.score(X, y)
 
 
+@pytest.mark.filterwarnings('ignore: default contamination parameter')  # 0.22
+def test_pipeline_score_samples_pca_lof():
+    iris = load_iris()
+    X = iris.data
+    # Test that the score_samples method is implemented on a pipeline.
+    # Test that the score_samples method on pipeline yields same results as
+    # applying transform and score_samples steps separately.
+    pca = PCA(svd_solver='full', n_components='mle', whiten=True)
+    lof = LocalOutlierFactor(novelty=True)
+    pipe = Pipeline([('pca', pca), ('lof', lof)])
+    pipe.fit(X)
+    # Check the shapes
+    assert_equal(pipe.score_samples(X).shape, (X.shape[0],))
+    # Check the values
+    lof.fit(pca.fit_transform(X))
+    assert_allclose(pipe.score_samples(X), lof.score_samples(pca.transform(X)))
+
+
+def test_score_samples_on_pipeline_without_score_samples():
+    iris = load_iris()
```

Comment:
For these kind of error checking tests, you don't need to use real-world data, or good parameters. We just need them to check what's needed, and to run fast.

`pipe = make_pipeline(LogisticRegression())` is enough

You could also just hard code X and y, or use `make_classification`

LGTM regardless, thanks @ab-anssi